File: app/src/main/java/com/chad/baserecyclerviewadapterhelper/adapter/SectionAdapter.java
package com.chad.baserecyclerviewadapterhelper.adapter;

import com.chad.baserecyclerviewadapterhelper.R;
import com.chad.baserecyclerviewadapterhelper.entity.MySection;
import com.chad.baserecyclerviewadapterhelper.entity.Video;
import com.chad.library.adapter.base.BaseSectionQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.List;

public class SectionAdapter extends BaseSectionQuickAdapter<MySection, BaseViewHolder> {

  /**
   * Same as QuickAdapter#QuickAdapter(Context,int) but with
   * some initialization data.
   *
   * @param sectionHeadResId The section head layout id for each item
   * @param layoutResId The layout resource id of each item.
   * @param data A new list is created out of this one to avoid mutable list
   */
  public SectionAdapter(int layoutResId, int sectionHeadResId, List data) {
    super(layoutResId, sectionHeadResId, data);
  }

  @Override
  protected void convertHead(BaseViewHolder helper, final MySection item) {
    helper.setText(R.id.header, item.header);
    helper.setVisible(R.id.more, item.isMore());
    helper.addOnClickListener(R.id.more);
  }

  @Override
  protected void convert(BaseViewHolder helper, MySection item) {
    Video video = (Video) item.t;
    switch (helper.getLayoutPosition() % 2) {
      case 0:
        helper.setImageResource(R.id.iv, R.mipmap.m_img1);
        break;
      case 1:
        helper.setImageResource(R.id.iv, R.mipmap.m_img2);
        break;

    }
    helper.setText(R.id.tv, video.getName());
  }
}

File: kotlinapp/src/main/java/com/allen/kotlinapp/adapter/SectionAdapter.kt
package com.allen.kotlinapp.adapter

import com.allen.kotlinapp.R
import com.allen.kotlinapp.entity.MySection
import com.allen.kotlinapp.entity.Video
import com.chad.library.adapter.base.BaseSectionQuickAdapter
import com.chad.library.adapter.base.BaseViewHolder

/**
 * 文 件 名: SectionAdapter
 * 创 建 人: Allen
 * 创建日期: 2017/6/14 18:53
 * 修改时间：
 * 修改备注：
 */
/**
 * Same as QuickAdapter#QuickAdapter(Context,int) but with
 * some initialization data.

 * @param sectionHeadResId The section head layout id for each item
 * *
 * @param layoutResId      The layout resource id of each item.
 * *
 * @param data             A new list is created out of this one to avoid mutable list
 */
class SectionAdapter constructor(layoutResId: Int, sectionHeadResId: Int, data: List<MySection>) : BaseSectionQuickAdapter<MySection, BaseViewHolder>(layoutResId, sectionHeadResId, data) {

    override fun convertHead(helper: BaseViewHolder, item: MySection) {
        helper.setText(R.id.header, item.header)
        helper.setVisible(R.id.more, item.isMore)
        helper.addOnClickListener(R.id.more)
    }


    override fun convert(helper: BaseViewHolder, item: MySection) {
        val video = item.t as Video
        when (helper.layoutPosition % 2) {
            0 -> helper.setImageResource(R.id.iv, R.mipmap.m_img1)
            1 -> helper.setImageResource(R.id.iv, R.mipmap.m_img2)
        }
        helper.setText(R.id.tv, video.name)
    }
}
